File: Swift-C++-Testsuite/test/basicOperators/C++/ComparisonString.cpp
#include <string>

bool stringEqual(std::string x) {
    bool tmp = x == "Hello";
    return tmp;
}

bool stringNotEqual(std::string x) {
    bool tmp = x != "Hello";
    return tmp;
}
bool stringLarger(std::string x) {
    bool tmp = x > "Hello";
    return tmp;
}
bool stringSmaller(std::string x) {
    bool tmp = x < "Hello";
    return tmp;
}
bool stringLargerOrEqual(std::string x) {
    bool tmp = x >= "Hello";
    return tmp;
}
bool stringSmallerOrEqual(std::string x) {
    bool tmp = x <= "Hello";
    return tmp;
}

int main() {
    bool a1 = stringEqual("Hello");
    bool a2 = stringEqual("World");
    bool b1 = stringNotEqual("Hello"); // false
    bool b2 = stringNotEqual("World"); // true
    bool c1 = stringLarger("Hello"); // true
    bool c2 = stringLarger("World"); //false
    bool d1 = stringSmaller("Hello"); // true
    bool d2 = stringSmaller("World"); // false
    bool e1 = stringLargerOrEqual("Hello"); // true
    bool e2 = stringLargerOrEqual("World"); // false
    bool f1 = stringSmallerOrEqual("Hello"); // true
    bool f2 = stringSmallerOrEqual("World"); // false
}

File: Swift-C++-Testsuite/test/basicOperators/swift/ComparisonString.swift
@main
struct MyMain {

    static func main() {
        var _ = stringEqual(x: "world")  // true
        var _ = stringEqual(x: "hello")  // false
        var _ = stringNotEqual(x: "world")  // false
        var _ = stringNotEqual(x: "hello")  // true
        var _ = stringLarger(x: "world")  // true
        var _ = stringLarger(x: "hello")  //false
        var _ = stringSmaller(x: "hello")  // true
        var _ = stringSmaller(x: "world")  // false
        var _ = stringLargerOrEqual(x: "world")  // true
        var _ = stringLargerOrEqual(x: "hello")  // false
        var _ = stringSmallerOrEqual(x: "world")  // true
        var _ = stringSmallerOrEqual(x: "hello")  // false
    }

    static func stringEqual(x: String) -> Bool {
        var tmp = x == "world"
        return tmp
    }
    static func stringNotEqual(x: String) -> Bool {
        var tmp = x != "world"
        return tmp
    }
    static func stringLarger(x: String) -> Bool {
        var tmp = x > "world"
        return tmp
    }
    static func stringSmaller(x: String) -> Bool {
        var tmp = x < "world"
        return tmp
    }
    static func stringLargerOrEqual(x: String) -> Bool {
        var tmp = x >= "world"
        return tmp
    }
    static func stringSmallerOrEqual(x: String) -> Bool {
        var tmp = x <= "world"
        return tmp
    }
}
